### PR TITLE
added problem 2182-1431-1768-151-1207

### PR DESCRIPTION
**Pull request by Sreedevi-27**

### CODE REVIEW

File: src/main/java/algorithms/leetcode/medium/GroupOfPeople.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Example 2:
+Input: groupSizes = [2,1,3,3,3,2]
+Output: [[1],[0,5],[2,3,4]]
+
+Constraints:
+groupSizes.length == n
+1 <= n <= 500
+1 <= groupSizes[i] <= n
+ */
+
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/medium/GroupOfPeople.java:31]: this problem can be refactored

File: src/main/java/algorithms/leetcode/medium/ReverseWordsInAString.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+There is at least one word in s.
+ */
+
+import java.util.Arrays;
+
+public class ReverseWordsInAString {
+    public static String reverseWords(String s){
+        String result = "";
+        String[] sArray = s.trim().split(" ");
+        System.out.println(Arrays.toString(sArray));
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/medium/ReverseWordsInAString.java:38]: remove this

File: src/main/java/algorithms/leetcode/medium/ReverseWordsInAString.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        int sArrayLength = sArray.length;
+
+        for(int i=sArrayLength-1; i>=0; i--){
+            if(!sArray[i].isEmpty())
+                result += sArray[i]+" ";
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/medium/ReverseWordsInAString.java:43]: use StringBuilder whenever you need to concatenate strings

File: src/main/java/algorithms/leetcode/medium/ReverseWordsInAString.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ */
+
+import java.util.Arrays;
+
+public class ReverseWordsInAString {
+    public static String reverseWords(String s){
+        String result = "";
+        String[] sArray = s.trim().split(" ");
+        System.out.println(Arrays.toString(sArray));
+        int sArrayLength = sArray.length;
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/medium/ReverseWordsInAString.java:39]: If the usage is only once then no need to store array length in variable use it directly

File: src/test/java/algorithms/leetcode/medium/GroupOfPeopleTest.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+class GroupOfPeopleTest {
+    @Test
+    void testGroupOfPeople(){
+        List<List<Integer>> expectedGroupPeople = new ArrayList<>();
+        List<Integer> list1 = Arrays.asList(5);
+        List<Integer> list2 = Arrays.asList(0,1,2);
+        List<Integer> list3 = Arrays.asList(3,4,6);
+        expectedGroupPeople.add(list1);
+        expectedGroupPeople.add(list2);
+        expectedGroupPeople.add(list3);
```
Comment (sharathr25) [inline src/test/java/algorithms/leetcode/medium/GroupOfPeopleTest.java:20]: you can do something like
```
List<List<Integer>> expectedGroupPeople = Arrays.asList(Arrays.asList(5), Arrays.asList(0,1,2), Arrays.asList(3,4,6))
```

**Review by sharathr25 — CHANGES_REQUESTED** [review]

File: src/test/java/algorithms/leetcode/medium/ReverseWordsInAStringTest.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package algorithms.leetcode.medium;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ReverseWordsInAStringTest {
+    void testReverseString(){
```
Comment (sharathr25) [inline src/test/java/algorithms/leetcode/medium/ReverseWordsInAStringTest.java:6]: Without @Test annotation how this test ran? Did you run the test?

File: src/main/java/algorithms/leetcode/easy/KidsWithGreaterCandies.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.List;
+
+public class KidsWithGreaterCandies {
+    public static List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
+        List<Boolean> isKidsHaveGreatestCandies = new ArrayList<>();
+        int maxCandy = 0;
+        for (int candy : candies)
+            maxCandy = Math.max(candy, maxCandy);
+        for (int candy : candies) {
+            boolean b = (isKidsHaveGreatestCandies.add(candy + extraCandies >= maxCandy));
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/easy/KidsWithGreaterCandies.java:49]: Whats this b. and where r u using this

File: src/main/java/algorithms/leetcode/easy/MergeStringAlternatively.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+Constraints:
+1 <= word1.length, word2.length <= 100
+word1 and word2 consist of lowercase English letters.
+ */
+public class MergeStringAlternatively {
+    public static String mergeAlternately(String word1, String word2) {
+        StringBuilder mergedString = new StringBuilder();
+        int word1Length = word1.length();
+        int word2Length = word2.length();
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/easy/MergeStringAlternatively.java:42]: word1.length() used only once. So u can directly use it. No need to create a variable. Same for word2.length(). 

Keep in mind if its used only once don't create a variable for it.

File: src/main/java/algorithms/leetcode/easy/MergeStringAlternatively.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        for(int i=0,j=0;  i<word1Length||j<word2Length; ){
+            mergedString.append(i < word1Length ? word1.charAt(i++) : "");
+            mergedString.append(j < word2Length ? word2.charAt(j++) : "");
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/easy/MergeStringAlternatively.java:45]: Only for true we need to append so use if condition.

File: src/main/java/algorithms/leetcode/easy/MergeStringAlternatively.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+Constraints:
+1 <= word1.length, word2.length <= 100
+word1 and word2 consist of lowercase English letters.
+ */
+public class MergeStringAlternatively {
+    public static String mergeAlternately(String word1, String word2) {
+        StringBuilder mergedString = new StringBuilder();
+        int word1Length = word1.length();
+        int word2Length = word2.length();
+        for(int i=0,j=0;  i<word1Length||j<word2Length; ){
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/easy/MergeStringAlternatively.java:43]: 👌

File: src/main/java/algorithms/leetcode/easy/UniqueOccurence.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.HashSet;
+import java.util.Map;
+
+public class UniqueOccurence {
+    public static  boolean uniqueOccurrences(int[] arr) {
+        Map<Integer,Integer> numberOccurences = new HashMap<>();
+        for(int element : arr)
+            numberOccurences.put(element,numberOccurences.getOrDefault(element,0)+1);
+        HashSet<Integer> uniqueElement = new HashSet<>(numberOccurences.values());
+        return (uniqueElement.size() == numberOccurences.size());
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/easy/UniqueOccurence.java:36]: 👌

File: src/main/java/algorithms/leetcode/medium/ReverseWordsInAString.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+
+import java.util.Arrays;
+
+public class ReverseWordsInAString {
+    public static String reverseWords(String s){
+        StringBuilder result = new StringBuilder();
+        String[] sArray = s.trim().split(" ");
+        for(int i=sArray.length-1; i>=0; i--){
+            if(!sArray[i].isEmpty())
+                result.append(sArray[i]).append(" ");
```
Comment (sharathr25) [inline src/main/java/algorithms/leetcode/medium/ReverseWordsInAString.java:40]: Here sArray[i] getting used twice. So u can store this in a variable.

**Review by sharathr25 — CHANGES_REQUESTED** [review]

File: src/main/java/algorithms/leetcode/medium/GroupOfPeople.java (hunk 1 above)
Comment (Sreedevi-27) [reply src/main/java/algorithms/leetcode/medium/GroupOfPeople.java:31]: Done ✅

**Review by sharathr25 — APPROVED** [review]

**Pull request merged**